account_payment_promissory_note/models/account_payment.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2018 Sergio Teruel <sergio.teruel@tecnativa.com>
+# Copyright 2018 Carlos Dauden <carlos.dauden@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountAbstractPayment(models.AbstractModel):
+    _inherit = 'account.abstract.payment'
+
+    deferred = fields.Boolean()
+    date_due = fields.Date(
+        string='Due Date',
+    )
+
+    @api.onchange('deferred')
+    def _onchange_deferred(self):
+        if not self.deferred:
+            self.date_due = False
+            return
+        invoices = False
+        if self._name == 'account.payment':
+            invoices = self.invoice_ids
+        elif self._name == 'account.register.payments':
```

Comment:
This mustn't be done here, but overwrite `_onchange_deferred` in the corresponding model, and perform this there.